frontend/src/store/channelStore.ts

```diff
@@ -1,35 +1,33 @@
-import { create } from 'zustand'
-
-interface Channel {
-  id: string;
-  name: string;
-  removable: boolean;
-}
+import { create } from 'zustand';
+import { Channel } from './interface';
 
 type Store = {
   allChannels: Channel[];
   currentChannel: Channel;
   currentChannelPopover: Channel;
   setCurrentChannel: (channel: Channel) => void;
-  setAllChannels: (allChannels: Channel[]) => void;
-  setNewChannel: (newChannel: Channel) => void;
+  setAllChannels: (channels: Channel[] | Channel) => void;
   setCurrentChannelPopover: (channel: Channel) => void;
-}
+};
 
 const channelStore = create<Store>()(
   (set, get) => ({
     allChannels: [],
-    currentChannel: {} as Channel,
-    currentChannelPopover: {} as Channel,
+    currentChannel: { id: '', name: '', removable: true },
+    currentChannelPopover: { id: '', name: '', removable: true },
```

Comment:
```suggestion
    currentChannel: defaultChannel,
    currentChannelPopover: defaultChannel,
```

defaultChannel можно объявить выше за пределами channelStore чтобы в случае изменения менять только в одном месте